Aave Funding Updates

efecarranza opened the pull request:
# Changelog

Add Ethereum mainnet proposal to deposit tokens into v3 and swap tokens for GHO.
Add Polygon proposal to bridge tokens to Ethereum mainnet.
Add tests for both proposals.
Add AIP.
Add deployment scripts.

rustboyar commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,47 @@
+- Redeem and transfer 0.5M DAI to Ethereum Treasury
+- Redeem and transfer 0.75M USDT to Ethereum Treasury
+
+Swap the following asset holdings to GHO:
```
ARC USDC missed here

efecarranza replied:
hey @rustboyar is there a tool to do the ARC transfer and withdrawal?
the owner of the funds to Transfer is GovernanceV3, while to withdraw from ARC, it's the ARC timelock.
I think I have to do it in 2 steps, a payload to transfer, and then a payload to withdraw, so was hoping maybe there's a way to do both?

efecarranza replied:
removing ARC 'cause it's just a painful process

Zer0dot commented on src/20231102_Multi_AaveFundingUpdates/AaveV2Ethereum_AaveFundingUpdates_20231102.t.sol:
```diff
@@ -0,0 +1,65 @@
+
+    GovHelpers.executePayload(vm, address(proposal), AaveGovernanceV2.ARC_TIMELOCK);
+
+    assertLt(
```
Any reason not to use something like `assertApproxEqAbs()` for more precision (with 1 wei rounding)?

efecarranza replied:
Removed ARC as it's just really painful to withdraw

Zer0dot commented on src/20231102_Multi_AaveFundingUpdates/AaveV2Ethereum_AaveFundingUpdates_20231102.t.sol:
```diff
@@ -0,0 +1,65 @@
+      balanceArcUSDCBefore
+    );
+
+    assertEq(IERC20(proposal.ARC_USDC()).balanceOf(address(proposal.SWAPPER())), 0);
```
Might be worth adding a check that the Swapper has the USDC itself no?

efecarranza replied:
hey @Zer0dot the swap isn't instantaneous unfortunately, it gets settled afterwards so no real way to settle that part i don't think.

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates_20231102.s.sol:
```diff
@@ -0,0 +1,74 @@
+
+    // create proposal
+    vm.startBroadcast();
+    GovV3Helpers.createProposal2_5(
```
```suggestion
    GovV3Helpers.createProposal2_5(
      vm,
```

sakulstra replied:
please accept the change and merge main to your branch

efecarranza commented:
@sakulstra @rustboyar @Zer0dot 
once again, apologies for the incomplete README, i have re read it and see how it was truly lacking.
i've updated the GHO oracle (how come the one in address book is the hardcoded one?) as well
i've also simplified the migration process, i did not ask, i only assumed since the tool existed that was the preferred approach to migrating.

sakulstra commented:
@efecarranza the address-book library is generated based on what is listed on aave.
For GHO the reality is that the `GHO` listed on aave is the minter and for minting GHO on the pool the price is considered to be exact 1, not taking into account secondary market prices.

https://app.aave.com/reserve-overview/?underlyingAsset=0x40d16fc0246ad3160ccc09b8d0d3a2cd28ae6c2f&marketName=proto_mainnet_v3

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+- Deposit 1M units of DAI from treasury (Aave V3 Ethereum Collector) into Aave v3 Ethereum
+- Migrate all aDAI from Aave v2 Ethereum to Aave v3 Ethereum
+
+- Redeem from Aave v2 Polygon 1.7M USDC and bridge to Ethereum treasury
```
```suggestion
- Redeem 1.7M USDC from Aave v2 Polygon and bridge to Ethereum treasury
```

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+- Migrate all aDAI from Aave v2 Ethereum to Aave v3 Ethereum
+
+- Redeem from Aave v2 Polygon 1.7M USDC and bridge to Ethereum treasury
+- Redeem from Aave v2 Polygon 0.5M DAI and bridge to Ethereum treasury
```
```suggestion
- Redeem 0.5M DAI from Aave v2 Polygon and bridge to Ethereum treasury
```

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+
+- Redeem from Aave v2 Polygon 1.7M USDC and bridge to Ethereum treasury
+- Redeem from Aave v2 Polygon 0.5M DAI and bridge to Ethereum treasury
+- Redeem from Aave v2 Polygon 0.75M USDT and bridge to Ethereum treasury
```
```suggestion
- Redeem 0.75M USDT from Aave v2 Polygon and bridge to Ethereum treasury
```

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+
+Swap the following asset holdings to GHO on Ethereum mainnet:
+
+- All available aTUSD that can be redeemed (57,373.85 units at time of writing)
```
```suggestion
- All available aTUSD that can be redeemed (~175,000 units at time of writing)
```

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+
+- All available aTUSD that can be redeemed (57,373.85 units at time of writing)
+- 500k units of DAI held in treasury (received from previous Polygon bridge)
+- All available aBUSD (~42,500 units at time of writing)
```
```suggestion
- All available aBUSD (~72,000 units at time of writing)
```

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+
+Swap the following asset holdings on Ethereum treasury to USDC (there is no GHO / ETH oracle thus these tokens cannot be swapped to GHO directly, they are Aave v2 Assets with no Aave v3 equivalent oracle):
+
+- All GUSD (18,464.03 units at time of writing)
```
```suggestion
- All GUSD (18,464.03)
```
it's underlying so won't change

sakulstra commented on src/20231102_Multi_AaveFundingUpdates/AaveFundingUpdates.md:
```diff
@@ -0,0 +1,48 @@
+Swap the following asset holdings on Ethereum treasury to USDC (there is no GHO / ETH oracle thus these tokens cannot be swapped to GHO directly, they are Aave v2 Assets with no Aave v3 equivalent oracle):
+
+- All GUSD (18,464.03 units at time of writing)
+- All UST (893,257.63 units at time of writing)
```
```suggestion
- All UST (893,257.63)
```
